src/agent/src/metrics.rs

```diff
@@ -84,23 +84,23 @@ pub fn get_metrics(_: &protocols::agent::GetMetricsRequest) -> Result<String> {
 
     let mut buffer = Vec::new();
     let encoder = TextEncoder::new();
-    encoder.encode(&metric_families, &mut buffer).unwrap();
+    encoder.encode(&metric_families, &mut buffer)?;
 
-    Ok(String::from_utf8(buffer).unwrap())
+    Ok(String::from_utf8(buffer)?)
 }
 
 #[instrument]
-fn update_agent_metrics() {
+fn update_agent_metrics() -> Result<()> {
     let me = procfs::process::Process::myself();
 
     if let Err(err) = me {
         error!(sl!(), "failed to create process instance: {:?}", err);
-        return;
+        return Ok(());
     }
 
-    let me = me.unwrap();
+    let me = me?;
```

Comment:
We just disposed of the error case above.  We can avoid both `unwrap()` and `?` if you combine the two cases into a `match`.
```
let me = match me {
    Ok(v) => v,
    Err(e) => {
        error!(sl!(), "failed to create process instance: {:?}", err);
        return Ok(());
}
```

Reply:
@dgibson @jodh-intel IMO, we need to avoid returning errors, that the `update_guest_metrics` after `update_agent_metrics` will get a chance to run, that we can get parts of metrics.

Some I think dgibson's suggestion can be moved to line 77:

```rust
match update_agent_metrics() {
    Ok(_) => {},
    Err(e) => { error!(sl!(), "failed to collect agent process metrics: {:?}", err);},
}
```

How about this?